xcode/Subconscious/Shared/Components/AddressBook/AddressBookModel.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  AddressBookModel.swift
+//  Subconscious (iOS)
+//
+//  Created by Ben Follington on 6/3/2023.
+//
+
+import os
+import Foundation
+import SwiftUI
+import ObservableStore
+
+struct AddressBookEntry: Equatable {
+    var pfp: Image
+    var petname: Petname
+    var did: Did
+}
+
+struct AddressBookEnvironment {
+    var noosphere: SphereIdentityProtocol
+}
+
+// Used for SwiftUI Previews, also useful for testing
+class PlaceholderSphereIdentityProvider: SphereIdentityProtocol {
+    func identity() throws -> String {
+        "did:key:z6MkmCJAZansQ3p1Qwx6wrF4c64yt2rcM8wMrH5Rh7DGb2K7"
+    }
+}
+
+enum AddressBookAction: Hashable {
+    case addFriend(did: Did, petname: Petname)
+    case removeFriend(did: Did)
+}
+
+struct AddressBookModel: ModelProtocol {
+    var friends: [AddressBookEntry] = []
```

Comment:
Nit: this might be better termed "follows"? Friends implies bi-directional to me.